lib/src/bitarray.dart

```diff
@@ -0,0 +1,317 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:collection' show ListMixin;
+import 'dart:typed_data' show Uint32List;
+import 'package:collection/collection.dart' show NonGrowableListMixin;
+
+/// A BitArray implementation for Dart to store boolean values.
+///
+/// Uses list of integers as internal storage to reduce memory usage.
+/// Implemented as fixed-length [List] of [bool].
+/// Throws error when calling any operation with length modifying.
+///
+/// Support several ways to perform logical operations (NOT, OR, AND, XOR):
+/// * instance methods (mutate [this]): [BitArray.not], [BitArray.or], [BitArray.and], [BitArray.xor];
```

Comment:
[this] -> `this`   
(The `this` is not a source name so you cannot link to it).
"mutate" -> "mutates" 
(present tense, not imperative).